### ISSUE TITLE
Make `pr_get_daynight` generic

### ISSUE THREAD

**Issue by jaseeverett** [PlanktonTeam/planktonr#116]
At the moment `pr_get_daynight` reloads the CPR data and doesn't work with NRS. I need to make this function more generic so that it can be used with any data frame that has a time.

**Comment by jaseeverett** [comment]
I have started work on this. At this stage I had created a new `pr_add_daynight` function and kept the old one till we can fully debug it and work out all the implications of replacing it.

**Comment by jaseeverett** [comment]
Due to the current "complete" command in the indices (and elsewhere) this leaves a lot of NAs for lat/lon which causes trouble with tz's. Need to discuss with @clairedavies how to deal with complete.

**Comment by clairedavies** [comment]
I disagree - we should only use the CPR data unless we have another big dataset that has day and night data. The NRS doesn't. We get into issues of non-comparable abundance data if we use more than one dataset and this isn't worth it when a dataset only has daytime data.
We also got rid of complete, so is your second comment still valid?

**Comment by jaseeverett** [comment]
I think you misunderstand and also I didn't explain very well! You are right. NRS is not useful. That was a bad example.

At the moment `pr_get_daynight` loads the CPR data from the AODN and calculates day night. This is silly if the user already has the CPR data in their workspace. It was mainly written for the BOO diurnal stuff.

For a user of planktonr (not BOO) this function would be much more useful if you load the CPR data using `pr_get_CPRData()` and pass it to `pr_get_daynight`. That way we could make it also work for any user's dataset that has a time and lat/lon column. Additionally, `pr_get_daynight` is hardwired for copepods (Z) or species (P) but users may want it for other groups as well. Again, ingesting a df into the function would make it more flexible. 

I started working on this here: https://github.com/PlanktonTeam/planktonr/blob/177ee56ffe3d70d0a5766fe09c04cd0cc770198a/R/utils_boo_data.R#L192
but haven't finished it yet. Will come back to it later.
